src/runtime/engines/imgbuild/engine.go

```diff
@@ -30,8 +30,8 @@ func (e *EngineOperations) Config() config.EngineConfig {
 	return e.EngineConfig
 }
 
-// CheckConfig validates EngineConfig setup
-func (e *EngineOperations) CheckConfig() error {
+// PrepareConfig validates/prepares EngineConfig setup
+func (e *EngineOperations) PrepareConfig() error {
 	e.CommonConfig.OciConfig.SetProcessNoNewPrivileges(true)
 
 	if syscall.Getuid() != 0 {
```

Comment:
Is this check for `syscall.Getuid() != 0` still necessary now that we have `IsAllowSUID()` function call? I added this check to error out if user is not root.

Reply:
Not necessary but that could help to display an explicit error message rather than fail later at container creation step during first mount